.circleci/config.yml

```diff
@@ -0,0 +1,51 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/python:3.5.5
+    working_directory: /home/circleci/msprime
+    steps:
+      - checkout
+      - restore_cache:
+          key: msprime-{{ .Branch }}
+      - run:
+          name: Install dependencies
+          command: |
+            sudo apt-get update
+            sudo apt-get install libgsl0-dev libhdf5-serial-dev libcunit1-dev libconfig-dev
+            pip install --user -r requirements/development.txt
+      - save_cache:
+          key: msprime-{{ .Branch }}
+          paths:
+            - "/home/circleci/.local"
+      - run:
+          name: Compile
+          command: |
+            CFLAGS=--coverage python setup.py build_ext --inplace
```

Comment:
Try "CC=h5cc CFLAGS=--coverage python setup.py build_ext --inplace" here.

h5cc should know where hdf5.h and the libs are.